### PR TITLE
App/gradio

### PR DESCRIPTION
**Pull request by deep-diver**
Here is the screenshot of the application (not Hugging Face Space friendly implementation yet - taking arguments from CLI). While experimenting with this, I found some issues. I will write them down here, so if you know the solution, please let me know. The issues will be patched with the future PR.

- when interacting with TGI served LLM
    - by setting `max_tokens` didn't guarantee to generate output that long (most of the time, just trimmed down shorter -> didn't even get complete paragraph)
    - always get the same response when regenerating (seems like get cached results back)

I guess it is because this issues are from when we use OpenAI SDK, which do not support `do_sample`, `use_cache`, ... This could be solved when switching back to use TGI client lib. Just wanted to check if you know about this issues.

overview

### CODE REVIEW

**Comment by sayakpaul** [comment]
> by setting max_tokens didn't guarantee to generate output that long (most of the time, just trimmed down shorter -> didn't even get complete paragraph)

I guess this issue should be raised on the TGI repository?

> This could be solved when switching back to use TGI client lib. 

Have you confirmed it?

We could maybe use ZeroGPU for the open access LLM part (use `transformers`) to have more control in case the TGI route doesn’t solve these issues (after confirming we are doing everything right).

**Comment by deep-diver** [comment]
At least it worked fine with naive approach: https://huggingface.co/spaces/chansung/llama2-with-gradio-chat

So I assume it would work with TGI client lib. Let me check.

**Comment by deep-diver** [comment]
@sayakpaul don't know about how to disable `use_cache` feature in TGI with OpenAI SDK, but I fixed `max_tokens` problem at least (found a bug in the code that I didn't pass the parameters to chat completion API)

Maybe in the HF Space, we can simply set GPU for this as you suggested.

For now, I think this PR could be reviewed and merged. WDYT?

File: vid2persona/gen/tgi_openllm.py [hunk 1]
```diff
@@ -6,17 +6,19 @@ async def send_messages(
     hf_token: str, 
     parameters: dict
 ):
-    parameters.pop('top_k')
-    parameters['max_tokens'] = parameters.pop('max_new_token')
-    parameters['presence_penalty'] = parameters.pop('repetition_penalty')
+    parameters.pop('repetition_penalty')
+    parameters['max_tokens'] = parameters.pop('max_new_tokens')
+    parameters['logprobs'] = True
+    parameters['top_logprobs'] = parameters.pop('top_k')
```
Comment (sayakpaul) [inline vid2persona/gen/tgi_openllm.py:12]: Perhaps, `parameters` could be validated using a decorator as well? `validate_tgi_parameters`? Not a blocker though!

**Review by sayakpaul — APPROVED** [review]
Just one comment. But this looks very solid overall!

**Pull request merged**